File: 274.hIndex.c
#include "shdstd.h"

int cmp32(const void *a, const void *b)
{
	return *(int *)b - *(int *)a;
}

int hIndex(int *citations, int size)
{
	qsort(citations, size, sizeof(int), cmp32);

	int h;
	for (h = 0; h < size; ++h) {
		if (citations[h] <= h) {
			break;
		}
	}

	return h;
}

File: 274.hIndex.go
package main

import "sort"

func hIndex(citations []int) int {
	sort.Slice(citations, func(left int, right int) bool {
		return citations[left] > citations[right]
	})

	h := 0
	for ; h < len(citations); h++ {
		if citations[h] <= h {
			break
		}
	}

	return h
}
